tools/mkbuildoptglobals.py

```diff
@@ -605,12 +611,64 @@ def parse_args():
     # ref nargs='*'', https://stackoverflow.com/a/4480202
     # ref no '--n' parameter, https://stackoverflow.com/a/21998252
 
+
+def get_encoding():
+    try:
+        from locale import getencoding
+    except ImportError:
+        # https://stackoverflow.com/a/23743499
+        return locale.getpreferredencoding(False) or locale.getdefaultlocale()[1]
+
+    return locale.getencoding()
```

Comment:
```suggestion
# retrieve *system* encoding, not the one used by python internally
if sys.version_info >= (3, 11):
    def get_encoding():
        return locale.getencoding()
else:
    def get_encoding():
        return locale.getdefaultlocale()[1]
```
?

no block scope in python
getpreferredencoding() would prefer (hm) utf-8 if we happen to use `-X utf-8`, behaviour changes depending on version. new one does not change with mode setting. both can return None aka default for us